jigglypuffRL/deeprl/dqn/dqn.py

```diff
@@ -0,0 +1,239 @@
+import math, random
+
+import gym
+import numpy as np
+
+import torch
+import torch.optim as opt
+from torch.autograd import Variable
+import torch.nn.functional as F
+
+import matplotlib.pyplot as plt
+
+from jigglypuffRL.common import ReplayBuffer, MlpValue
+
+
+class DQN:
+    """
+    Deep Q Networks 
+    Paper: (DQN) https://arxiv.org/pdf/1312.5602.pdf
+    Paper: (Double DQN) https://arxiv.org/abs/1509.06461
+    :param network_type: (str) The deep neural network layer types ['MLP']
+    :param env: (Gym environment) The environment to learn from
+    :param double_dqn: (boolean) For training Double DQN
+    :param epochs: (int) Number of epochs
+    :param max_iterations_per_epoch: (int) Number of iterations per epoch
+    :param max_ep_len: (int) Maximum steps per episode
+    :param gamma: (float) discount factor
+    :param lr: (float) learing rate for the optimizer 
+    :param batch_size: (int) Update batch size
+    :param replay_size: (int) Replay memory size
+    :param plot_loss_reward_graph: (bool) To print the loss and reward as a graph or text
+    :param tensorboard_log: (str) the log location for tensorboard (if None, no logging)
+    :param seed : (int) seed for torch and gym
+    :param render : (boolean) if environment is to be rendered
+    :param device : (str) device to use for tensor operations; 'cpu' for cpu and 'cuda' for gpu
+    """
+
+    def __init__(
+        self,
+        network_type,
+        env,
+        double_dqn=False,
+        epochs=100,
+        max_iterations_per_epoch=100,
+        max_ep_len=1000,
+        gamma=0.99,
+        lr=0.001,
+        batch_size=32,
+        replay_size=100,
+        plot_loss_reward_graph=False,
```

Comment:
Loss and reward plots will be on tensorboard anyways right? Is this required then?